### PR TITLE
Enable automatic syncing with upstream

### PR DESCRIPTION
**Pull request by merlijn-sebrechts**
To make it easier to keep this snap up to date, I propose to add this action that updates the repo automatically when a new release is detected. (this automatically runs daily)

So the new workflow is:

* Action detects a new version and commits it
* Store detects changes and rebuilds to edge
* We test and manually push to stable

This should make my job a lot easier. What do you think? The main downside I see is that I have to give myself permission to override branch protections (or create a new bot account specially for this?)

PS: In the future, I'd like the building and publishing to also happen using actions, so we can publish to `candidate` instead, and I can get a notification when it's ready for testing.

### CODE REVIEW

**Comment by galgalesh** [comment]
Another option is to keep the logic that downloads the latest version and trigger the snap store's "rebuild" logic when I see the version in the store is different from the latest release. However, I think having the version in the repository is a better approach because it gives us more reproducibility and it allows us to rebuild the snap without getting the latest version of the app itself.

File: .github/workflows/sync-release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Fetch release version
+        run: |
+          VERSION=$(
+            curl -sL https://api.github.com/repos/signalapp/Signal-Desktop/releases | 
+            jq .  | grep tag_name | grep -v beta | head -n 1 | cut -d'"' -f4 | tr -d 'v'
+          )
+          sed -i 's/^\(version: \).*$/\1'"$VERSION"'/' snap/snapcraft.yaml
+      - name: Check for modified files
+        id: git-check
+        run: echo ::set-output name=modified::$([ -z "`git status --porcelain`" ] && echo "false" || echo "true")
```
Comment (zyga) [inline .github/workflows/sync-release.yml:26]: IIRC this is now deprecated.

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

It should be replaced with:
```
modified=... >> $GITHUB_OUTPUT
```

**Comment by galgalesh** [comment]
Fixed, but builds are now failing because of ubuntu repos being down..

**Review by diddledani — CHANGES_REQUESTED** [review]
We can probably use `plugin: dump`:

```yaml
  parts:
    signal-desktop:
      plugin: dump
      source: https://updates.signal.org/desktop/apt/pool/main/s/signal-desktop/signal-desktop_$SNACPRAFT_PROJECT_VERSION_amd64.deb
      build-packages:
        - gcc
        - wget
        - jq
        - dpkg
        - git
      stage-packages:
        - libxss1
        - libnspr4
        - libnss3
        - libvips42
      override-build: |
        set -xeu
        craftctl default
        sed -i 's|Icon=signal-desktop|Icon=${SNAP}/usr/share/icons/hicolor/512x512/apps/signal-desktop.png|' $CRAFT_PART_INSTALL/usr/share/applications/signal-desktop.desktop
      prime:
        - opt
        - usr/share
        - usr/lib/*/libxss*
        - usr/lib/*/*nss*
        - usr/lib/*/libvips*
        - usr/lib/*/libI*
        - usr/lib/*/libMagic*
        - usr/lib/*/libcfit*
        - usr/lib/*/libcgif*
        - usr/lib/*/libexif*
        - usr/lib/*/libfftw*
        - usr/lib/*/libgsf*
        - usr/lib/*/libheif*
        - usr/lib/*/libimage*
        - usr/lib/*/libmat*
        - usr/lib/*/libopensli*
        - usr/lib/*/libwebp*
        - usr/lib/*/libHalf*
        - usr/lib/*/libaom*
        - usr/lib/*/libdav1d*
        - usr/lib/*/libx265*
        - usr/lib/*/libhdf*
        - usr/lib/*/libsz*
        - usr/lib/*/liblqr*
        - usr/lib/*/libaec*
        - usr/lib/*/libde265*
        - usr/lib/*/libnuma*
        - -opt/Signal/chrome-sandbox
        - -opt/Signal/resources/app.asar.unpacked/node_modules/sharp/vendor/lib
        - -opt/Signal/resources/app.asar.unpacked/node_modules/sharp/vendor/include
```

File: .github/workflows/sync-release.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  schedule:
+    - cron:  '0 10 * * *'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  get-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (merlijn-sebrechts) [reply .github/workflows/sync-release.yml]: Thanks, fixed!

**Comment by merlijn-sebrechts** [comment]
Thanks, @diddledani , I greatly cleaned up the logic. The dump plugin does everything correctly for us.

**Review by diddledani — APPROVED** [review]
LGTM. For a followup, it looks like snapcore/action-build#42 is now fixed so we can try to revert to the `ubuntu-latest` runner once this is merged.

**Comment by merlijn-sebrechts** [comment]
Thanks, I'll try it in the next refactor. I'll merge this PR, rename this branch to `candidate`, and work on the auto-publishing logic.

**Pull request merged**